### PR TITLE
PORTAL-3383 Adapt TypeScript and upgrade documentation generator

### PR DESCRIPTION
**Pull request by toruta39**
https://rgames.atlassian.net/browse/PORTAL-3383

- [x] typescript
- [x] documentation generator
- [x] flowtype clean up

### CODE REVIEW

**Comment by toruta39** [comment]
@Zack0711 just finished revamping type definitions from flowtype to typescript. the source code part are ready for review.

File: src/sdk/payments.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  PaymentsGetCatalogResponse,
+  PaymentsPurchaseResponse,
+  PaymentsGetPurchasesResponse,
+  PaymentsConsumePurchaseResponse
+} from '../types/bridge';
+
+/**
+ * (Experimental)
+ * @example
+ * ViberPlay.payments.onReady(function () {
```
Comment (Zack0711) [inline src/sdk/payments.ts]: Missing wrapping ``` to add code blocks for example.

File: src/sdk/player.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  CanSubscribeBotResponse,
+  SubscribeBotResponse,
+} from '../types/bridge';
+
+/**
+ * Will get an object representing game data saved on the hosted platform.
+ * @param keys - An array of unique keys to retrieve data for.
+ * @returns Latest snapshot of game data
+ * @example
+ * ViberPlay.player
```
Comment (Zack0711) [inline src/sdk/player.ts]: Missing wrapping ``` to add code blocks for example.

File: src/types/context.ts [hunk 3]
```diff
@@ -34,9 +33,29 @@ export type ContextChooseFilter =
...
+  maxSize?: number;
+  minSize?: number;
+  hoursSinceInvitation?: number;
+}
+
+/**
+ * @typedef {Object} ContextSizeResponse
+ * @property {boolean} answer - Result
+ * @property {number} minSize - The minimum bound of the context size query
+ * @property {number} maxSize - The maximum bound of the context size query.
```
Comment (Zack0711) [inline src/types/context.ts]: Missing to remove @typedef and @property .

File: src/models/ad-instance.ts [hunk 4]
```diff
@@ -28,7 +33,7 @@ export default class AdInstance {
    * @example
    * adInstance.getPlacementID(); // '5458282176661711'
```
Comment (Zack0711) [inline src/models/ad-instance.ts]: Missing wrapping ``` to add code blocks for example.

**Pull request merged**